### DISCUSSION TITLE
How safe is my network from VPN server?

### DISCUSSION THREAD

**Discussion by urwrstkn8mare** [qdm12/gluetun#1290] [Q&A] [answered]
Hi, 

Just wondering how safe I would be if my glue tun container was connected to a compromised VPN server? Would it be able to access my wider network or do any damage to my systems, OS or network through the container? 

Thanks,
Samit

**Comment by qdm12** [comment, marked as answer]
Hi there,

This is a good question that took me some time to read on 😄

Please correct me if I'm wrong anywhere in the following or something feels off;

The Wireguard network interface acts really like an ethernet cable between the two peers. Now if the server (one of the two peers) gets compromised they can send any network packet they want through the tunnel. So really it's up to the other peer to limit what it can do.

In Gluetun's case, the firewall by default blocks all inbound traffic from the Wireguard interface. Note traffic initialized from gluetun (or connected containers etc.) going through the Wireguard interface is allowed so data replied can be garbage, but here again the application which sent the data would notice this (bad tls validation etc.).
So at least the compromised wireguard server cannot access other services/ports listening on the gluetun container (such as the http control server).

I don't believe there is any way it could access a shell in gluetun (which has root access and NET_ADMIN so it would be bad if it could).

**Reply by urwrstkn8mare** [reply]
Thanks alot for the detailed answer and for making this container - it really goes far in providing peace of mind.

btw I'm assuming this is the same as in openvpn

**Reply by qdm12** [reply]
I think so, at least for the tun device we use here, not sure for the tap device mode but it's not used in gluetun.

Just more thoughts, if you send unencrypted and unverified traffic (for example plain http to an ip address), a compromised server would be able to act as man in the middle and return you any data it wants, and you would accept it. But that's also true for any network hop machine up to your destination (without vpn or with un compromised vpn server), so really it would be your fault 😸 That's why we have TLS (for https, dns etc.) 😉

**Reply by urwrstkn8mare** [reply]
Thanks!

**Comment by urwrstkn8mare** [comment]
Thanks!
